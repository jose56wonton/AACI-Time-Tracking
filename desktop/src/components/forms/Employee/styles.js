const styles = () => ({
  gridContainer: {
    width: '100% !important',
    margin: 0
  },
  row: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  field:{
    margin: 'auto 8px'
  },
  error: { marginTop: '16px' },
});

export default styles;
