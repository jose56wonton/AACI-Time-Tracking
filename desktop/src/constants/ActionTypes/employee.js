export const UPDATE_EMPLOYEE_REQUEST = 'update_employee_request';
export const UPDATE_EMPLOYEE_SUCCESS = 'update_employee_success';
export const UPDATE_EMPLOYEE_FAILURE = 'update_employee_failure';

export const GET_EMPLOYEE_REQUEST = 'get_employee_request';
export const GET_EMPLOYEE_SUCCESS = 'get_employee_success';
export const GET_EMPLOYEE_FAILURE = 'get_employee_failure';

export const CREATE_EMPLOYEE_REQUEST = 'create_employee_request';
export const CREATE_EMPLOYEE_SUCCESS = 'create_employee_success';
export const CREATE_EMPLOYEE_FAILURE = 'create_employee_failure';

export const DELETE_EMPLOYEE_REQUEST = 'delete_employee_request';
export const DELETE_EMPLOYEE_SUCCESS = 'delete_employee_success';
export const DELETE_EMPLOYEE_FAILURE = 'delete_employee_failure';

export const LOGIN_EMPLOYEE_REQUEST = 'login_employee_request';
export const LOGIN_EMPLOYEE_SUCCESS = 'login_employee_success';
export const LOGIN_EMPLOYEE_FAILURE = 'login_employee_failure';

export const CLOCKIN_EMPLOYEE_REQUEST = 'clockin_employee_request';
export const CLOCKIN_EMPLOYEE_SUCCESS = 'clockin_employee_success';
export const CLOCKIN_EMPLOYEE_FAILURE = 'clockin_employee_failure';
export const CLOCKIN_EMPLOYEE_SNACK = 'clockin_employee_snack';

export const CLOCKOUT_EMPLOYEE_REQUEST = 'clockout_employee_request';
export const CLOCKOUT_EMPLOYEE_SUCCESS = 'clockout_employee_success';
export const CLOCKOUT_EMPLOYEE_FAILURE = 'clockout_employee_failure';
export const CLOCKOUT_EMPLOYEE_SNACK = 'clockout_employee_snack';
