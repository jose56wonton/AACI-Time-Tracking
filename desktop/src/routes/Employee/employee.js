import accountActionScene from 'scenes/Account/accountAction';
import clockOutScene from 'scenes/Clock/clockOut';
import * as routes from 'constants/routes';

export default [
  {
    path: `/${routes.EMPLOYEE}`,
    component: accountActionScene,
    exact: true,
    name: `Employee Actions`,
    type: routes.EMPLOYEE
  },
  {
    path: `/${routes.EMPLOYEE}/${routes.CLOCKOUT}`,
    component: clockOutScene,
    exact: true,
    name: `Employee Clockout`,
    type: routes.EMPLOYEE
  }
];
