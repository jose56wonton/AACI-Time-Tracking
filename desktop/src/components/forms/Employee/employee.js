import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {  Grid,  Typography,  Button,  Tooltip,  IconButton } from '@material-ui/core';
import cx from 'classnames';
import { Field, Form } from 'formik';
import { withStyles } from '@material-ui/core/styles';
import { Delete } from '@material-ui/icons';

import TextField from 'components/inputs/TextField';
import Select from 'components/inputs/Select';
import Switch from 'components/inputs/Switch';
import styles from './styles';

class Employee extends Component {
  render() { 
    const { classes, crews, authorities, label, isSubmitting, type, deleteEmployee, resetForm, initialValues,errors } = this.props;
    //console.log(this.props);
    return (
      <Form>
        <Grid container spacing={24} className={classes.gridContainer}>
          <Grid item xs={12} className={classes.row}>
            <Typography variant="h6">{label}</Typography>
            {type === 'edit' && (
              <Tooltip title="Delete">
                <IconButton onClick={deleteEmployee} aria-label="Delete">
                  <Delete />
                </IconButton>
              </Tooltip>
            )}
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <Field
              name="firstName"
              component={TextField}
              margin="none"
              label="First Name"
              type="search"
              className={classes.field}
              helper="normal"
            />
            <Field
              name="lastName"
              component={TextField}
              margin="none"
              label="Last Name"
              type="search"
              className={classes.field}
              helper="normal"
            />
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <Field
              name="authorityId"
              component={Select}
              items={authorities}
              fullWidth
              label="Authority"
              className={classes.field}
            />
            <Field
              name="crewId"
              component={Select}
              items={crews}
              fullWidth
              label="Crew"
              className={classes.field}
            />
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <div className={cx(classes.field,classes.switchBox)}>
              <Field
                name="isEmployed"
                component={Switch}
                label="Is Employed"
                className={classes.field}
              />
              <Field
                name="isWorking"
                component={Switch}
                disabled={true}
                label="Is Working"
                className={classes.field}
              />
            </div>

            <Field
              name="pin"
              component={TextField}
              margin="none"
              label="Pin"
              type="search"
              className={classes.field}
              helper="normal"
            />
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <Typography
              color="error"
              variant="button"
              className={classes.field}
            >
              {errors.submit}
            </Typography>
            <div>
              <Button
                type="submit"
                color="primary"
                disabled={isSubmitting}
                variant="contained"
                className={classes.button}
              >
                Save
              </Button>
              <Button
                onClick={() => {
                  resetForm(initialValues);
                }}
                disabled={isSubmitting}
                color="secondary"
                variant="text"
                className={classes.button}
              >
                Reset
              </Button>
            </div>
          </Grid>
        </Grid>
      </Form>
    );
  }
}

Employee.propTypes = {
  classes: PropTypes.object.isRequired,
  authorities: PropTypes.array.isRequired,
  crews: PropTypes.array.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  deleteEmployee: PropTypes.func,
  resetForm: PropTypes.func.isRequired,
  initialValues: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};

export default withStyles(styles)(Employee);
