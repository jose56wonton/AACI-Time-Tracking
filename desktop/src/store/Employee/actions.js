import { normalize } from 'normalizr';
import moment from 'moment';

import { employeeActionTypes } from 'constants/ActionTypes';
import { shiftActions, snackActions, activityActions } from 'store/actions';
import * as endpoint from './endpoints';
import * as schemas from 'store/schemas';
import * as status from 'constants/status';
import { currentRoundedTime } from 'helpers/time';

export const getEmployees = () => {
  return async dispatch => {
    dispatch({ type: employeeActionTypes.GET_EMPLOYEE_REQUEST });
    try {
      const response = await endpoint.getEmployees();
      const payload = normalize(
        { employees: response.data },
        schemas.employeeArray,
      );
      return dispatch({
        type: employeeActionTypes.GET_EMPLOYEE_SUCCESS,
        payload
      });
    } catch (e) {
      console.log(e);
      return dispatch({
        type: employeeActionTypes.GET_EMPLOYEE_FAILURE,
        payload: e
      });
    }
  };
};

export const putEmployee = employee => {
  return async dispatch => {
    dispatch({ type: employeeActionTypes.UPDATE_EMPLOYEE_REQUEST });
    try {
      const response = await endpoint.putEmployee(employee.id, employee);
      const payload = normalize(
        { employees: [response.data] },
        schemas.employeeArray,
      );
      return dispatch({
        type: employeeActionTypes.UPDATE_EMPLOYEE_SUCCESS,
        payload
      });
    } catch (e) {
      console.log(e);
      return dispatch({
        type: employeeActionTypes.UPDATE_EMPLOYEE_FAILURE,
        payload: e
      });
    }
  };
};

export const toggleIsWorking = employee => {
  return putEmployee({
    ...employee,
    isWorking: !employee.isWorking
  });
};

export const clockIn = employee => {
  return async dispatch => {
    dispatch({ type: employeeActionTypes.CLOCKIN_EMPLOYEE_REQUEST });
    try {
      const clockInObject = {
        clockInDate: currentRoundedTime()
          .subtract(7, 'hours')
          .toString(),
        employeeId: employee.id
      };
      await dispatch(shiftActions.postShift(clockInObject));
      await dispatch(toggleIsWorking(employee));
      await dispatch(
        snackActions.openSnack(status.SUCCESS, 'Clock in Success!'),
      );
      return dispatch({ type: employeeActionTypes.CLOCKIN_EMPLOYEE_SUCCESS });
    } catch (e) {
      console.log(e);
      dispatch(snackActions.openSnack(status.FAILURE, 'Clock in failed!'));
      return dispatch({
        type: employeeActionTypes.CLOCKIN_EMPLOYEE_FAILURE,
        payload: e
      });
    }
  };
};

export const clockOut = (employee, shift, activities) => {
  return async dispatch => {
    dispatch({ type: employeeActionTypes.CLOCKOUT_EMPLOYEE_REQUEST });
    try {
      const currentMoment = currentRoundedTime();
      const clockInMoment = moment(shift.clockInDate);
      const shiftDuration = moment.duration(currentMoment.diff(clockInMoment));
      const minutes = shiftDuration.asMinutes();

      const clockOutObject = {
        ...shift,
        clockOutDate: currentMoment.toString(),
        length: minutes
      };
      await activities.forEach(activity => {
        // activity.projectId = undefined;
        activity.shiftId = shift.id;
        dispatch(activityActions.postActivity(activity));
      });

      await dispatch(shiftActions.putShift(clockOutObject));
      await dispatch(toggleIsWorking(employee));
      dispatch(snackActions.openSnack(status.SUCCESS, 'Clock out Success!'));
      return dispatch({ type: employeeActionTypes.CLOCKOUT_EMPLOYEE_SUCCESS });
    } catch (e) {
      console.log(e);
      dispatch(snackActions.openSnack(status.SUCCESS, 'Clock out Success!'));
      return dispatch({
        type: employeeActionTypes.CLOCKOUT_EMPLOYEE_FAILURE,
        payload: e
      });
    }
  };
};

export const login = pin => {
  return async dispatch => {
    dispatch({ type: employeeActionTypes.LOGIN_EMPLOYEE_REQUEST });
    try {
      const response = await endpoint.getEmployeeByPin(pin);
      const payload = normalize(
        { employees: [response.data] },
        schemas.employeeArray,
      );

      return dispatch({
        type: employeeActionTypes.LOGIN_EMPLOYEE_SUCCESS,
        payload,
        data: response.data
      });
    } catch (e) {
      dispatch({
        type: employeeActionTypes.LOGIN_EMPLOYEE_FAILURE,
        payload: e
      });
      throw e;
    }
  };
};
