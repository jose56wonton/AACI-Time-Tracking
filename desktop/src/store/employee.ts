import {
  ActivityState,
  Activity,
  BaseEmployee,
  BaseActivity,
  BaseShift
} from './types';
import { snackActions } from 'store/actions';
import { AxiosError } from 'axios';
import { Action, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import axios from 'helpers/axios';
import moment from 'moment';
import * as status from 'constants/status';

export interface ClockInEmployeeRequest
  extends Action<'ClockInEmployeeRequest'> {}
export interface ClockInEmployeeFailure
  extends Action<'ClockInEmployeeFailure'> {
  error: AxiosError;
}
export interface ClockInEmployeeSuccess
  extends Action<'ClockInEmployeeSuccess'> {}

export interface ClockOutEmployeeRequest
  extends Action<'ClockOutEmployeeRequest'> {}
export interface ClockOutEmployeeFailure
  extends Action<'ClockOutEmployeeFailure'> {
  error: AxiosError;
}
export interface ClockOutEmployeeSuccess
  extends Action<'ClockOutEmployeeSuccess'> {}

export type EmployeeActions =
  | ClockInEmployeeRequest
  | ClockInEmployeeFailure
  | ClockInEmployeeSuccess
  | ClockOutEmployeeRequest
  | ClockOutEmployeeFailure
  | ClockOutEmployeeSuccess;

export const clockIn = (
  employee: BaseEmployee
): ThunkAction<void, BaseEmployee, null, ClockInEmployeeSuccess> => {
  return async (dispatch: Dispatch<EmployeeActions>) => {
    dispatch({ type: 'ClockInEmployeeRequest' });
    try {
      console.log(employee.id);
      await axios.post('/employees/clockin', {
        employeeId: employee.id
      });
      dispatch(snackActions.openSnack(status.SUCCESS, `Clock in success!`) as any);
      dispatch({ type: 'ClockInEmployeeSuccess' });
    } catch (e) {
      dispatch({ type: 'ClockInEmployeeFailure', error: e });
      throw e;
    }
  };
};

export const clockOut = (
  employee: BaseEmployee,
  shift: BaseShift,
  activities: BaseActivity[],
  lunch: number
): ThunkAction<void, BaseEmployee, null, ClockInEmployeeSuccess> => {
  return async (dispatch: Dispatch<EmployeeActions>) => {
    dispatch({ type: 'ClockOutEmployeeRequest' });
    try {
      const shiftRequest = {
        lunch: lunch,
        length: moment.duration(moment().diff(moment(shift.clockInDate))).asMinutes()
      };

      console.log(employee, shift, shiftRequest, activities, lunch);
      await axios.post('/employees/clockout', {
        employeeId: employee.id,
        shift: shiftRequest,
        activities: activities
      });
      dispatch({ type: 'ClockOutEmployeeSuccess' });
    } catch (e) {
      dispatch({ type: 'ClockOutEmployeeFailure', error: e });
      throw e;
    }
  };
};

