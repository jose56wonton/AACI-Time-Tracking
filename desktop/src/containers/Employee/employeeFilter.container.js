import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Formik } from 'formik';

import Employee from 'components/forms/Employee';
import { authoritySelectors, crewSelectors } from 'store/selectors';
import { analyzeActions } from 'store/actions';

import domain from 'constants/domains';

export class EmployeeFilter extends Component {  
  removeEmployee = () => {
    const { selected, removeEmployee } = this.props;  
    removeEmployee(selected.id);
  };

  render() {
    const { authorities, crews,employeeFilters  } = this.props;
    
    return (
      <Formik
        enableReinitialize
        initialValues={{
          ...employeeFilters
        }}
        onSubmit={(values, formikFunctions) => {
          const { updateFilter } = this.props;
          updateFilter({
            ...values
          });
          formikFunctions.resetForm();          
        }}
        render={formikProps => {
          return (
            <Employee
              authorities={authorities}
              crews={crews}
              label="Filter"
              type="filter"
              {...formikProps}
            />
          );
        }}
      />
    );
  } 
      
    
  
}

/* istanbul ignore next */
const mapStateToProps = state => {
  return {
    employeeFilters: state.analyze.employeeFilters,
    crews: crewSelectors.getAllCrews(state),
    authorities: authoritySelectors.getAllAuthorities(state)
  };
};

/* istanbul ignore next */
const mapDispatchToProps = dispatch => {
  return {
    updateFilter: filters => dispatch(analyzeActions.updateFilter(domain.EMPLOYEE,filters))
  };
};

export default connect(mapStateToProps,mapDispatchToProps)(EmployeeFilter);
