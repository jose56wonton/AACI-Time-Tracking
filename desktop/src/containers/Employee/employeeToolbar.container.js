import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { analyzeActions } from 'store/actions';
import { employeeSelectors } from 'store/selectors';
import Progress from 'components/helpers/Progress';
import { analyzeStatus } from 'constants/analyze';
import domain from 'constants/domains';


import AnalyzeToolbar from 'components/tables/toolbars/NewAnalyzeToolbar';


export class EmployeeIndex extends Component { 
  
  selectLabel = selected => `${selected.firstName} ${selected.lastName} selected`;

  select = object => this.props.select(domain.EMPLOYEE,object)

  add = () => {
    const { selected, select, setStatus } = this.props;
    if(selected && selected.id){
      select(domain.EMPLOYEE,selected.id);
    }
    setStatus(domain.EMPLOYEE,analyzeStatus.ADDING);
  } 
  
  render() {
    const { employees, selected, toggleEmployeeFilter,employeeFilterVisible } = this.props;
    // console.log(toggleEmployeeFilter);

    // if employees is not defined or empty
    if (!employees) return <Progress variant="circular" fullPage />;

    return (   
      <div >        
        <AnalyzeToolbar 
          selectLabel={this.selectLabel}
          label="Employees"
          add={this.add}
          toggleFilter={toggleEmployeeFilter}
          isFilterVisible={employeeFilterVisible}
          selected={selected}
        />        
      </div>              
    );
  }
}

EmployeeIndex.propTypes = {
  employees: PropTypes.array,
  selected: PropTypes.object,
  select: PropTypes.func.isRequired,
  setStatus: PropTypes.func.isRequired,
  toggleEmployeeFilter: PropTypes.func,
  employeeFilterVisible: PropTypes.bool
};

/* istanbul ignore next */
const mapStateToProps = state => {

  const filters = state.analyze.employeeFilters;
  return {
    employeeFilterVisible: state.analyze.employeeFilterVisible,
    employees: employeeSelectors.getAllEmployeesNew(state, { filters , sorts: {} }),
    selected: employeeSelectors.getSelectedEmployee(state)
  };
};

/* istanbul ignore next */
const mapDispatchToProps = dispatch => {
  return {
    ...bindActionCreators({ ...analyzeActions }, dispatch),    
    toggleEmployeeFilter: () => dispatch(analyzeActions.toggleFilter(domain.EMPLOYEE))  
  };
};

export default connect(mapStateToProps,mapDispatchToProps)(EmployeeIndex);
