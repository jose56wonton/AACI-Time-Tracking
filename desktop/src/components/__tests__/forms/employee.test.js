import React from 'react';
import { shallow } from 'enzyme';

import { Employee } from 'components/forms/Employee/employee';
import EmployeeHOC from 'components/forms/Employee';

const props =  {  
  classes: {},
  crews: [{},{}],
  authorities: [{},{}],
  type: 'edit',
  removeEmployee: jest.fn(),
  editCrews: jest.fn(),
  editAuthorities: jest.fn(),
  label: 'label',
  isSubmitting: true,
  resetForm: jest.fn(),
  initialValues: {},
  errors: {}
};

const setup = overRides => {  
  return shallow(<Employee {...props} {...overRides}/>);    
};

const setupHOC = overRides => {
  return shallow(<EmployeeHOC {...props} {...overRides}/>);
};

describe.skip('Employee Component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call resetForm on employee-reset-button', () => {
    const wrapper = setup();
    const instance = wrapper.instance();
    instance.resetForm = jest.fn();
    expect(props.resetForm).toHaveBeenCalledTimes(0);
    wrapper.find('#employee-reset-button').first().simulate('click');
    expect(props.resetForm).toHaveBeenCalledTimes(1);
  });
});