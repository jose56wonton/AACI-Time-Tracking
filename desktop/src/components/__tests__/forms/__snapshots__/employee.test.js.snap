// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Component should render correctly 1`] = `
<Form>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      className=""
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        variant="h6"
      >
        label
      </WithStyles(Typography)>
      <WithStyles(Tooltip)
        title="Delete"
      >
        <WithStyles(IconButton)
          aria-label="Delete"
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <C
        component={[Function]}
        helper="normal"
        label="First Name"
        margin="none"
        name="firstName"
        type="search"
      />
      <C
        component={[Function]}
        helper="normal"
        label="Last Name"
        margin="none"
        name="lastName"
        type="search"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <C
        component={[Function]}
        fullWidth={true}
        items={
          Array [
            Object {},
            Object {},
          ]
        }
        label="Authority"
        name="authorityId"
      />
      <WithStyles(Tooltip)
        title="Edit Authorities"
      >
        <WithStyles(IconButton)
          aria-label="Edit Authorities"
          onClick={[Function]}
        >
          <pure(EditIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
      <C
        component={[Function]}
        fullWidth={true}
        items={
          Array [
            Object {},
            Object {},
          ]
        }
        label="Crew"
        name="crewId"
      />
      <WithStyles(Tooltip)
        title="Edit Crews"
      >
        <WithStyles(IconButton)
          aria-label="Edit Crews"
          onClick={[Function]}
        >
          <pure(EditIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <div
        className=""
      >
        <C
          component={[Function]}
          label="Is Employed"
          name="isEmployed"
        />
        <C
          component={[Function]}
          label="Is Working"
          name="isWorking"
        />
      </div>
      <C
        component={[Function]}
        helper="normal"
        label="Pin"
        margin="none"
        name="pin"
        type="search"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        color="error"
        variant="button"
      />
      <div>
        <WithStyles(Button)
          color="primary"
          disabled={true}
          type="submit"
          variant="contained"
        >
          Save
        </WithStyles(Button)>
        <WithStyles(Button)
          color="secondary"
          disabled={true}
          id="employee-reset-button"
          onClick={[Function]}
          variant="text"
        >
          Reset
        </WithStyles(Button)>
      </div>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Form>
`;

exports[`Employee Component should render correctly withStyles 1`] = `
<Employee
  authorities={
    Array [
      Object {},
      Object {},
    ]
  }
  classes={
    Object {
      "button": "Employee-button-7",
      "error": "Employee-error-6",
      "field": "Employee-field-4",
      "gridContainer": "Employee-gridContainer-1",
      "headerRow": "Employee-headerRow-3",
      "row": "Employee-row-2",
      "switchBox": "Employee-switchBox-5",
    }
  }
  crews={
    Array [
      Object {},
      Object {},
    ]
  }
  editAuthorities={[Function]}
  editCrews={[Function]}
  errors={Object {}}
  initialValues={Object {}}
  isSubmitting={true}
  label="label"
  removeEmployee={[Function]}
  resetForm={[Function]}
  type="edit"
/>
`;
